core/tee/tee_fs_key_manager.c

```diff
@@ -403,5 +403,58 @@ TEE_Result tee_fs_decrypt_file(enum tee_fs_file_type file_type,
 			cipher, cipher_size, plaintext, plaintext_size);
 }
 
+static void u16_to_big_endian(uint16_t u16, uint8_t *bytes)
+{
+	*bytes = (uint8_t) (u16 >> 8);
+	*(bytes + 1) = (uint8_t) u16;
+}
+
+/*
+ * AES CTR mode encryption/decryption used for block encryption of RPMB FS
+ * files.
+ */
+TEE_Result tee_fs_crypt_block(uint8_t *out, const uint8_t *in, size_t size,
+			      uint16_t blk_idx, const uint8_t *encrypted_fek,
+			      const uint8_t *nonce, TEE_OperationMode mode)
+{
+	TEE_Result res;
+	uint8_t fek[TEE_FS_KM_FEK_SIZE];
+	uint8_t iv[TEE_AES_BLOCK_SIZE];
+	uint8_t *ctx;
+	size_t ctx_size;
+
+	DMSG("%scrypt block #%u", (mode == TEE_MODE_ENCRYPT) ? "En" : "De",
+	     blk_idx);
+
+	memcpy(fek, encrypted_fek, TEE_FS_KM_FEK_SIZE);
+	res = fek_crypt(TEE_MODE_DECRYPT, fek, TEE_FS_KM_FEK_SIZE);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = crypto_ops.cipher.get_ctx_size(TEE_ALG_AES_CTR, &ctx_size);
+	if (res != TEE_SUCCESS)
+		return res;
+	ctx = malloc(ctx_size);
+	if (!ctx)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	memcpy(iv, nonce, 14);
```

Comment:
Agreed. But, I will change this part of the code to use AES CBC with ESSIV instead (see the discussion we had in the doc thread: https://github.com/OP-TEE/optee_os/pull/631#discussion_r51059024.
So the nonce will be removed.